private/pkg/filepathext/filepathext_windows.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build windows
+// +build windows
+
+package filepathext
+
+import (
+	"os"
+	"path/filepath"
+)
+
+var (
+	// USERPROFILE shows the current user's profile folder. It is being used to parse the volume
+	// for the current user and this is used as the process's context.
+	// https://learn.microsoft.com/en-us/windows/deployment/usmt/usmt-recognized-environment-variables#variables-that-are-processed-for-the-operating-system-and-in-the-context-of-each-user
+	FSRoot = filepath.VolumeName(os.Getenv("USERPROFILE")) + string(os.PathSeparator)
```

Comment:
I retract my comment earlier, I think it isn't feasible to parse this through the system env vars -- the process could just be running in an entirely separate drive, which is what is happening in the actions. We should use `normalpath.Components` when parsing the absolute paths and document clearly the behaviours of how `reader.go` is constructing the OS root buckets.